File: services/notification-service/app/events/event_metadata.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class EventMetadata(BaseModel):
    model_config = ConfigDict(
        populate_by_name=True
    )

    event_id: UUID = Field(alias="eventId")
    correlation_id: UUID | None = Field(alias="correlationId")
    occurred_at: datetime = Field(alias="occurredAt")
    event_type: str = Field(alias="eventType")
    event_version: int = Field(alias="eventVersion")

File: common/atlanta-common-events/src/main/java/com/atlanta/common/events/metadata/EventMetadata.java
package com.atlanta.common.events.metadata;

import com.atlanta.common.events.constants.ServiceName;

import java.time.Instant;
import java.util.UUID;

public record EventMetadata(

        UUID eventId,

        UUID correlationId,

        Instant occurredAt,

        ServiceName source,

        String eventName,

        int eventVersion

) {
}